oauth2/oauth2-core/src/main/java/org/springframework/security/oauth2/oidc/core/user/DefaultOidcUser.java

```diff
@@ -13,67 +13,68 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+
 package org.springframework.security.oauth2.oidc.core.user;
 
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
 import org.springframework.security.core.GrantedAuthority;
 import org.springframework.security.oauth2.core.user.DefaultOAuth2User;
 import org.springframework.security.oauth2.oidc.core.IdToken;
 import org.springframework.security.oauth2.oidc.core.IdTokenClaim;
-import org.springframework.security.oauth2.oidc.core.StandardClaim;
 import org.springframework.security.oauth2.oidc.core.UserInfo;
-
-import java.util.Map;
-import java.util.Set;
-import java.util.stream.Collectors;
-import java.util.stream.Stream;
-
-import static org.springframework.security.oauth2.oidc.core.StandardClaim.NAME;
+import org.springframework.util.Assert;
 
 /**
  * The default implementation of an {@link OidcUser}.
  *
  * <p>
  * The claim used for accessing the &quot;name&quot; of the
  * user <code>Principal</code> via {@link #getClaims()}
- * is {@link StandardClaim#NAME} or if not available
- * will default to {@link IdTokenClaim#SUB}.
+ * is {@link IdTokenClaim#SUB}.
  *
  * @author Joe Grandja
+ * @author Vedran Pavic
  * @since 5.0
  * @see OidcUser
  * @see DefaultOAuth2User
  * @see IdToken
  * @see UserInfo
  */
 public class DefaultOidcUser extends DefaultOAuth2User implements OidcUser {
-	private final IdToken idToken;
```

Comment:
I don't feel we should get rid of these fields. Even though the claims from both `IdToken` and `UserInfo` are copied to `attributes` we may want to provide getters for these 2 constructs.